source/auth/auth.rst

```diff
@@ -486,8 +486,8 @@ Second, the client MAY implement its own canonicalization. If so, the canonicali
     # Unspecified which CNAME is used if > 1.
     host = one of the records in cnames
 
-  reversed = do a reverse DNS lookup for address
-  if reversed:
+  if forwardAndReverse:
+    reversed = do a reverse DNS lookup for address
```

Comment:
`CANONICALIZE_HOST_NAME` is currently equivalent to `forwardAndReverse`. This PR changes it to mean `forward`. Is that the correct behaviour? Why are we potentially breaking backward compat?

Reply:
As well the Python impl below does not take `true` versus `forward` versus `forwardAndReverse` into account. It still implements the old pseudo code where only `true` was applicable and had the same meaning as `forwardAndReverse`.

Reply:
Given that we are asking drivers to implement (or wrap libgssapi) to canonicalize host names, do we want to add some host name canonicalization tests to ensure that drivers are performing canonicalization correctly? It seems dangerous to leave this behaviour testing up to each driver individually as it may give rise to differences in implementation.

Reply:
Good catch. I was basing this off the Node implementation which doesn't do the reverse lookup so I'll update to make forwardAndReverse the default. Also yes I will add some more tests as well and ping you when the changes are made.

Reply:
I've updated to default `true` to being "forwardAndRevers", updated the python impl, and added spec test for raising an error on invalid values.

Reply:
The Java driver calls `[java.net.InetAddress#getByName(java.lang.String)](https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getByName-java.lang.String-)`, followed by `[java.net.InetAddress#getCanonicalHostName](https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getCanonicalHostName--)`.  I think that's the equivalent of forwardAndReverse.... Can you confirm?

Reply:
Yes that is the equivalent of forwardAndReverse.